geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -419,6 +419,23 @@ impl IntersectionMatrix {
         && self.0[CoordPos::Outside][CoordPos::OnBoundary] == Dimensions::Empty
     }
 
+    /// Returns `true` if every point in Geometry `b` is a point of Geometry `a` interior.
+    ///
+    /// # Notes
+    /// - If Geometry `b` has any interaction with the boundary of Geometry `a`, then the result is `false`.
+    /// - Geometry`a` will never contains_properly itself.
+    /// - Matches `[T**FF*FF*]`
+    /// - This predicate is **transitive** but not **reflexive**
```

Comment:
Could you please cross reference the two implementations in their corresponding docs.

Also, it'd be good to mention something about how performance characteristics might change between them.

If you have an intuition for when it's preferable to use `ContainsProperly` vs `Relate`, please include that in the docs too.  As you mentioned, the baton rouge test is a lot slower when `ContainsProperly`. Is it that generally `Relate` works better for larger inputs?

Reply:
Which part of the docs should I put that in?
1. Trait (top level of doc)
2. Trait impl (in the docs on a specific impl)
3. fn in trait (not in docs but shows up in tooling when using the fn)